src/Module/Item/Ignored.php

```diff
@@ -43,10 +44,20 @@ public static function rawContent()
 		// Numeric values are needed for the json output further below
 		$ignored = !empty($thread['ignored']) ? 0 : 1;
 
-		if (!empty($thread['uid']) && $thread['uid'] != 0) {
-			$dba->update('thread', ['ignored' => $ignored], ['iid' => $message_id]);
-		} else {
-			$dba->update('user-item', ['ignored' => $ignored], ['iid' => $message_id, 'uid' => local_user()], true);
+		switch ($thread['uid'] ?? 0) {
+			// if the thread is from the current user
+			case local_user():
+				$dba->update('thread', ['ignored' => $ignored], ['iid' => $message_id]);
+				break;
+			// Empty or 0 (null will get transformed to 0) => it's a public post
+			case 0:
+			case '':
+				$dba->update('user-item', ['ignored' => $ignored], ['iid' => $message_id, 'uid' => local_user()], true);
+				break;
+			// In case we retrieved a thread which isn't our or a public, it's a forbidden action
+			// but due to security reason (brute force), we print a Bad request exception
```

Comment:
What's the security reason between forbidden and bad request?

Reply:
That's what I assumed. Can you please be more specific about preventing enumeration in the comment?

Alternatively, use item guids?